### PR TITLE
rlpx: parse auth message and send ack

### PR DESCRIPTION
**Pull request by lechengfan**
* Parse the auth handshake message to extract the init nonce, remote pub key, and remote ephemeral pub keys
* Send an ack message in response to that
* Finish up tests for the handshake

### CODE REVIEW

File: rlpx/handshake.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package rlpx
 import (
 	"crypto/rand"
 	"errors"
-	mrand "math/rand"
+	// mrand "math/rand"
```
Comment (lechengfan) [inline rlpx/handshake.go:6]: Commenting this out for the time being since RLP can't seem to decode padded messages.

Comment (ryandotsmith) [reply rlpx/handshake.go:6]: I'm not so sure about this. I just tested the RLP package and I wasn't able to reproduce. Decode will return an error if the length of the input doesn't match the length encoded in the body. I also successfully decoded `rlp.List(rlp.Bytes([]byte{0x01, 0x00, 0x00}))`

Is this PR in a state where I can reproduce the bug? I un-commented out the padding code on this branch, but I got a seemingly unrelated error. Is that error in fact what you are referring to?

File: rlpx/handshake.go [hunk 2]
```diff
@@ -112,18 +99,15 @@ func (h *handshake) createAckMsg() (rlp.Item, error) {
 	if h.isInitiator {
 		return rlp.Item{}, errors.New("cannot send ack message when you are the initiator")
 	}
-	if h.receiverNonce != nil {
-		_, err = rand.Read(h.receiverNonce[:])
-		if err != nil {
-			return rlp.Item{}, err
-		}
+	h.receiverNonce = make([]byte, 32)
```
Comment (ryandotsmith) [inline rlpx/handshake.go:102]: ditto

**Review by ryandotsmith — APPROVED** [review]
lgtm!

**Pull request merged**